src/Universum/Monad/Either.hs

```diff
@@ -82,24 +86,46 @@ maybeToRight l = maybe (Left l) Right
 maybeToLeft :: r -> Maybe l -> Either l r
 maybeToLeft r = maybe (Right r) Left
 
--- | Applies given action to 'Either' content if 'Left' is given.
-whenLeft :: Applicative f => Either l r -> (l -> f ()) -> f ()
-whenLeft (Left  l) f = f l
-whenLeft (Right _) _ = pass
+-- | Applies given action to 'Either' content if 'Left' is given and returns
+-- the result. In case of 'Right' the default value will be returned.
+whenLeft :: Applicative f => Either l r -> a -> (l -> f a) -> f a
```

Comment:
Usually such functions expect default argument as first argument. So it's more convenient and expected for users to have type like this:

```haskell
whenLeft :: Applicative f => a -> Either l r -> (l -> f a) -> f a
```